urlcaef331f4e_dreamable_cra_tgz-pJ8-coho_lp_project_LPProjectJ8/src/coho/lp/project/DoubleIntervalLPProject.java

```
@@ -0,0 +1,308 @@
+						// backtrack till find the minimum angle.						
+						LPBasis newB = lpSolver.opt((x.add(y.mult(maxInc)))).optBasis();//direction=x+y*tan(maxAngle)
+						if(!newB.equals(B)){
+							DoubleIntervalMatrix point = (DoubleIntervalCohoMatrix.create(A.row(B.basis()),false)).getSolution(b.V(B.basis()));								
+							DoubleIntervalMatrix newPoint = (DoubleIntervalCohoMatrix.create(A.row(newB.basis()),false)).getSolution(b.V(newB.basis()));
+							double length =newPoint.sub(point).norm().doubleValue(); 
+							//we may skip edge shorter than minLength
+							if(length>minLength){//should not jump to next quad
+								//if(DEBUG.debug) DEBUG.println(LPproject_DEBUG,"LPproject.findPlanes(): There is a new basis "+newB.toString()+" in the same quadrant.");
+								Iterator<LPBasis> iter = result.optBasesIter();
+								try{
+									while(!B.equals(iter.next())){
+									}//until we find the current basis
+									if(iter.hasNext()){//if contains the right endpoint
+										//backtrackTimes=0;//reset
+										B = iter.next();//use the next basis										
+										//if(DEBUG.debug) DEBUG.println(LPproject_DEBUG,"LPproject.findPlanes(): Let's try another possible optimal basis "+B.toString());
+									}else{//force to the right endpoint
+										//XXX: here, we don't update alpha now because we update backtrackTimes
+										//result = rightEnd(alpha,currNorm.E(),B,result,solver,A,x,y,false,true,8*backtrackTimes);//last para can be true or false. result.bestOptBasis==B
+										//at least once, at most 5 times.
+										result = rightEnd(alpha,currNorm.e().doubleValue(),B,result,lpSolver,A,x,y,8*backtrackTimes,1,5);//last para can be true or false. result.bestOptBasis==B
+										B = result.optBasis();
+										backtrackTimes++;
+										//if(DEBUG.debug) DEBUG.println(LPproject_DEBUG,"LPproject.findPlanes(): Let's try an new basis "+B.toString());
+									}						
+								}catch(NoSuchElementException e){
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)